Sources/SwiftGRPCNIO/GRPCClientChannelHandler.swift

```diff
@@ -104,6 +107,7 @@ internal class GRPCClientChannelHandler<RequestMessage: Message, ResponseMessage
   ///
   /// - Parameter error: the error to observe.
   internal func observeError(_ error: GRPCError) {
+    self.errorDelegate?.didCatchError(error.error, file: error.file, line: error.line)
```

Comment:
When exactly is this method called, could you refresh my memory?
(Might also need to update the documentation for this method now.)

Reply:
At the moment it's called by `errorCaught(context:error:)` in the final handler in the pipeline. It's also called when the call times out and when the call is cancelled.

I have a follow up PR to change those last two usages into user events (i.e. handled with `userInboundEventTriggered(context:event:)`). This would stop `observeError` from being called outside of this class.

Reply:
Makes sense. Are there any other _additional_ places where we could/should call this? (Not sure myself; if you say that this is the only place, I trust you.)

Reply:
It should be called in handlers at the end of the NIO pipeline. In this case we have the HTTP/2 multiplexed channels and pipelines (i.e. in the snippet above) and behind the multiplexer for errors that aren't at the stream level.